src/crud.js

```diff
@@ -0,0 +1,191 @@
+    saveEdit.remove();
+    deleteEdit.remove();
+  }
+  taskList.classList.remove('active');
+  inputId.setAttribute('readonly', true);
+};
+
+const saveTodos = (e) => {
+  const saveBtn = e.target;
+  const btnClass = saveBtn.className;
+  const btnId = btnClass.split('-');
+  const id = parseInt(btnId[1], 10);
+  saveTodo(id, true);
+};
+
+const editTodos = (e) => {
+  const editInput = e.target;
+  const editBtn = document.getElementById(`${editInput.id}`);
+  const inputId = document.querySelector(`#activity-${editInput.id}`);
+  inputId.removeAttribute('readonly');
+  inputId.focus();
+
+  editBtn.style.display = 'none';
+  inputId.style.outlineColor = 'gray';
+
+  const taskList = document.querySelector(`#tasks-${editInput.id}`);
+  taskList.classList.add('active');
+
+  const task = document.querySelector(`#task-${editInput.id}`);
+
+  const save = document.createElement('img');
+  save.setAttribute('src', saveIcon);
+  save.classList.add(`save-${editInput.id}`);
+  task.appendChild(save);
+
+  const removeBtn = document.createElement('img');
+  removeBtn.classList.add(`delete-${editInput.id}`);
+  removeBtn.setAttribute('src', deleteIcon);
+  task.appendChild(removeBtn);
+
+  document.querySelectorAll(`.delete-${editInput.id}`).forEach((e) => {
+    e.addEventListener('click', deleteTodos);
+  });
+
+  document.querySelectorAll(`.save-${editInput.id}`).forEach((e) => {
+    e.addEventListener('click', saveTodos);
+  });
+};
+
+const showToDos = () => {
+  const existingTodos = JSON.parse(localStorage.getItem('todos'));
+  if (existingTodos !== null && existingTodos.length > 0) {
+    const collection = document.querySelector('.todo-collection');
+    collection.innerHTML = '';
+
+    existingTodos.forEach((todo, index) => {
+      const unordered = document.createElement('ul');
+      const ordered = document.createElement('li');
+      const liner = document.createElement('hr');
+
+      unordered.classList.add('tasks');
+      unordered.id = `tasks-${index}`;
+
+      ordered.classList.add('tasks');
+      ordered.id = `task-${index}`;
+
+      const check = document.createElement('input');
+      check.type = 'checkbox';
+      check.classList.add('check');
+      check.id = `check-${index}`;
+
+      const input = document.createElement('input');
+      input.type = 'text';
+      input.classList.add('activity');
+      input.id = `activity-${index}`;
+      input.value = `${todo.description}`;
+      if (todo.completed) {
+        check.checked = true;
+        input.classList.add('completed');
+      }
+
+      const edit = document.createElement('img');
+      edit.setAttribute('src', dots);
+      edit.classList.add('edit');
+      edit.id = index;
+
+      collection.appendChild(unordered);
+      unordered.appendChild(ordered);
+      unordered.appendChild(liner);
+      ordered.appendChild(check);
+      ordered.appendChild(input);
+      ordered.appendChild(edit);
+
+      ['focus', 'blur', 'keyup'].forEach((evt) => {
```

Comment:
Please consider storing all these events in a separate variable at the top of the file so as to be able to re-use it on line 163 as well